hadoop-hdds/framework/src/main/java/org/apache/hadoop/ozone/common/BlockGroup.java

```diff
@@ -30,24 +29,41 @@ public final class BlockGroup {
 
   private String groupID;
   private List<BlockID> blockIDs;
-
-  private BlockGroup(String groupID, List<BlockID> blockIDs) {
+  private List<Long> blockSizes;
+  private List<Long> replicatedBlockSizes;
+  private List<DeletedBlock> deletedBlocks;
+  public static final long SIZE_NOT_AVAILABLE = -1;
+
+  private BlockGroup(String groupID,
+                     List<BlockID> blockIDs,
+                     List<Long> blockSizes,
+                     List<Long> replicatedBlockSizes,
+                     List<DeletedBlock> deletedBlocks) {
```

Comment:
I'm not sure why deletedBlocks is required here. It looks like in each case,  BlockGroup either have deletedBlocks, or have blockIDs, blockSizes and replicatedBlockSizes.

Reply:
Please add an check to verify that blockIDs, blockSizes, replicatedBlockSizes have the same length.